### ISSUE TITLE
refract_extended

### ISSUE THREAD

**Issue by vreijs** [rstub/swephR#22]
Added refract_extended: (I wa snot able to push to master...)
this code in swephR.cpp
```
//' Compute the refraction
//' @param InAlt  The object's apparent/topocentric altitude (depending on calc_flag)
//' @param geoheight  The observer's height
//' @param lapse_rate  The lapse rate pK/m]
//' @param calc_flag SE_TRUE_TO_APP=0 or SE_APP_TO_TRUE=1
//' @return \code{swe_refrac_extended} returns a list with named entries: \code{i} success of function
//'      \code{dret} for refraction related calculations (TopoAlt, AppAlt, refraction)
//' @rdname expert-interface
//' @export
// [[Rcpp::export(swe_refrac_extended)]]
Rcpp::List refrac_extended(double InAlt, double geoheight, double atpress, double attemp, double lapse_rate, int calc_flag) {
  std::array<double, 10> dret{0.0};
  int i = swe_refrac_extended(InAlt,geoheight,atpress,attemp,lapse_rate,calc_flag, &dret[0]);
  return Rcpp::List::create(Rcpp::Named("return") = i,
                            Rcpp::Named("dret") = dret);
}

```

and this code in test-misc.R:
```
test_that("Topocentric to Apparent altitude", {
  result <- swe_refrac_extended(2,0,1013.25,15,-0.065,0)
  expect_equal(result$return, 2)
  expect_equal(result$dret[1:3], c(2.0000000, 2.2787088, 0.2787088), tolerance=.000001)
})

test_that("Apparent to Topocentric altitude", {
  result <- swe_refrac_extended(2,0,1013.25,15,-0.065,1)
  expect_equal(result$return, 1)
  expect_equal(result$dret[1:3], c(1.701016, 2.000000, 0.298984), tolerance=.000001)
})

I think that with this one, I have also all skyscpeR functions. Hope this helps.
```

**Comment by rstub** [comment]
Please use the new branch https://github.com/rstub/swephR/tree/vreijs/functions4 for pushing this commit. But note that `swephR.cpp` has been split into multiple files. Thanks.

**Comment by vreijs** [comment]
Done, add swe_refraction_extended.
 It seems I still need to do:

> roxygen2::roxygenise()Writing NAMESPACELoading swephRWriting NAMESPACE
Writing calculation.Rd
Writing expert-interface.Rd
Writing eclipse_pheno.Rd
Writing swephR-package.Rd


>

to get the new function (swe_refraction_extended) in the list of functions.

**Comment by rstub** [comment]
Thanks. Note that I have moved this function to eclipse_pheno.cpp, since it is documented in section 6 together with the other functions from that file: http://www.astro.com/swisseph/swephprg.htm?lang=g#_Hlk477329626 

With this commit also the CI should be green. So if you have nothing else in the pipeline, I would go ahead and merge that branch.

**Comment by vreijs** [comment]
At this moment I have nothing else.
What is now needed ot get it on CRAN. The feedback form the SE author is
not very fast (one of them is/was serious ill), but I think we 'solved' all
the major warnings (it would though be nice to see their acknowledgement
and their ack that they will implement the changes).
So I asked feedback on:
. swe_close(): we have a work around for this (keep setting swe_set_topo).
. dobs array length: minimally 4 elements and initialised.
. warnings in Windows: they are only minor
. warnings in Clang: C-code workaround available
. warning in Debian: C-code workaround available
. swe_phone_ut behavior: workaround found (If that is always correct, will
need to hear from SE authors).
. copyright string: lets use "Authors of the Swiss Ephemeris" and install
the LICENSE.TXT along with the package (both the R-code and the SWIEPH
data).

I now am now concentrating on ARCHAEOCOSMO and skyscapeR. It is mainly
skyscapeR where I need to determine how to link with swephR: without
changing the UI of swephR. I might meet the author of skyscpeR this
Saturday at a conference on:
https://www.eventbrite.ie/e/pathways-to-the-cosmos-tickets-45383757039

By the way, your input and work is very great in progressing the
development of swephR!!! Thanks very much for that!!! By the way, why are
you so dedicated on this subject? It is sometimes not easy to encounter
such people.

**Comment by vreijs** [comment]
Sorry, stop. I will add the swe_azalt_rev() in the coming minutes. I think that is still a handy one for skyscapeR

**Comment by vreijs** [comment]
Ok I now added the swe_azalt_rev function (which can be handy for skyscapeR(). So I think I am now really ready to only work on skyscapeR;-)

**Comment by rstub** [comment]
> I now am now concentrating on ARCHAEOCOSMO and skyscapeR. It is mainly skyscapeR where I need to determine how to link with swephR: without changing the UI of swephR.

... but if changes are necessary, we can incorporate them.

> I might meet the author of skyscpeR this Saturday at a conference on: https://www.eventbrite.ie/e/pathways-to-the-cosmos-tickets-45383757039

Interesting. At one point I would like to get a better understanding what your research is like.

> By the way, why are you so dedicated on this subject?

A combination of different things: In my youth I was fascinated by both archeology and astronomy, but never thought about combining the two. Now, after a PhD in physics and several years of consulting work I am doing a lot of work with R. And I (again) enjoy the open source world very much. And I enjoy answering questions on Stack Overflow ....

**Comment by rstub** [comment]
This has been addressed in #23.
